packages/teleport-generator-shared/src/utils/project-utils.ts

```diff
@@ -248,14 +249,144 @@ export const createFile = (name: string, fileType: string, content: string): Gen
   return { name, fileType, content }
 }
 
-export const createFolder = (
-  name: string,
-  files: GeneratedFile[] = [],
-  subFolders: GeneratedFolder[] = []
+interface LocalDependenciesMeta {
+  defaultComponentsPath: string | string[]
+  defaultPagesPath: string | string[]
+}
+
+export const generateLocalDependenciesPrefix = (
+  template: TemplateDefinition,
+  meta: LocalDependenciesMeta
+): string => {
+  const initialComponentsPath =
+    template.meta && template.meta.componentsPath
+      ? template.meta.componentsPath
+      : [].concat(meta.defaultComponentsPath)
+
+  const initialPagesPath =
+    template.meta && template.meta.pagesPath
+      ? template.meta.pagesPath
+      : [].concat(meta.defaultPagesPath)
+
+  let dependencyPrefix = ''
+
+  /*
+    Remove common path elements from the beginning of the
+    components and pages full path (if any)
+  
+    For example, having:
+    - initialComponentsPath = ['src', 'components']
+    - initialPagesPath = ['src', 'pages']
+  
+    If we want to have an import statement that goes from the pages folder to the
+    components folder, we only need to go back one step, so we are removing
+    the forst element from both the paths ('src') and build the dependencyPrefix accordingly
+  */
+  const { componentsPath, pagesPath } = removeCommonStartingPointsFromPaths(
+    initialComponentsPath,
+    initialPagesPath
+  )
+
+  // We have to go back as many folders as there are defined in the pages path
+  dependencyPrefix += '../'.repeat(pagesPath.length)
+
+  dependencyPrefix += componentsPath
+    .map((component) => {
+      return `${component}/`
+    })
+    .join('')
+
+  return dependencyPrefix
+}
+
+const removeCommonStartingPointsFromPaths = (
+  componentsPath: string[],
+  pagesPath: string[]
+): { componentsPath: string[]; pagesPath: string[] } => {
+  const componentsPathLength =
+    componentsPath.length > pagesPath.length ? componentsPath.length : pagesPath.length
+
+  let i = 0
+  let isEqual = true
+  while (i < componentsPathLength && isEqual) {
+    if (componentsPath[i] === pagesPath[i]) {
+      componentsPath.shift()
+      pagesPath.shift()
+    } else {
+      isEqual = false
+    }
+    i += 1
+  }
+
+  return { componentsPath, pagesPath }
+}
+
+export const injectFilesToPath = (
+  rootFolder: GeneratedFolder,
+  path: string[],
+  files: GeneratedFile[]
 ): GeneratedFolder => {
+  let folder = findFolderByPath(rootFolder, path)
+
+  if (!folder) {
+    const { updatedRootFolder, createdFolder } = createFolderByPath(rootFolder, path)
+    rootFolder = updatedRootFolder
+    folder = createdFolder
+  }
+
+  folder.files = folder.files.concat(files)
```

Comment:
you can do folder.files.push(files)

Reply:
Well, actually I can't. The **folder.files** entity is an array of **GeneratedFiles** and the **files** that I'm trying to inject into the already existing array is also of type **GeneratedFiles[]**. If I would use the push method I would get a matrix (array of array) not a flat structure, like I need.